packages/react/src/Dialog/Dialog.module.css

```diff
@@ -1,3 +1,16 @@
+/* The --dialog-scrollgutter property is used only on the body element to
+ * simulate scrollbar-gutter:stable. This property is not and should not
+ * be used elsewhere in the DOM. There is a performance penalty to
+ * setting inherited properties which can cause a large style recalc to
+ * occur, so it benefits us to prevent inheritance for this property.
+ * See https://web.dev/blog/at-property-performance
+ */
+@property --dialog-scrollgutter {
+  initial-value: 0;
+  inherits: false;
+  syntax: '<length>';
+}
+
```

Comment:
Most of this implementation was taken from what we did in PVC https://github.com/primer/view_components/pull/2542